integration-test/scripts/swagger-check.sh

```diff
@@ -51,160 +86,31 @@ IFS=,
-# if the size of the non-API change collector array is equal with the number of changed files
-# then all the changed files are in non-API modules
-# and swagger check can be skipped
-if [[ ${#nonApiFileChanges[@]} -eq ${#changedFiles[@]} ]]; then
-  echo "All the changes are in non-API modules so swagger check can be skipped"
-  exit 0
-fi
-
 for service in $Services; do
   if [ "${CB_TARGET_BRANCH}" != "master" ] && verlte 2.31.0-b118 $PREVIOUS_BUILD ; then
-    echo Downloading ${service} ${PREVIOUS_BUILD} swagger definition, if possible
-    STATUSCODE=$(curl -kfSs --write-out "%{http_code}" https://${service}-swagger.s3.${zone[$service]}.amazonaws.com/swagger-${PREVIOUS_BUILD}.json -o ./apidefinitions/${service}-swagger-${PREVIOUS_BUILD}.json)
-    if [ $STATUSCODE -ne 200 ]; then
-      echo download failed $STATUSCODE
-      rm ./apidefinitions/${service}-swagger-${PREVIOUS_BUILD}.json
+    if ! compatible "${service}" "${PREVIOUS_BUILD}"; then
+      INCOMPATIBLE_CHANGES+=("$service:$PREVIOUS_BUILD")
     fi
   fi
   if verlte 2.31.0-b118 $PREVIOUS_MINOR_BUILD ; then
-    echo Downloading ${service} ${PREVIOUS_MINOR_BUILD} swagger definition, if possible
-    STATUSCODE=$(curl -kfSs --write-out "%{http_code}" https://${service}-swagger.s3.${zone[$service]}.amazonaws.com/swagger-${PREVIOUS_MINOR_BUILD}.json -o ./apidefinitions/${service}-swagger-${PREVIOUS_MINOR_BUILD}.json)
-    if [ $STATUSCODE -ne 200 ]; then
-      echo download failed $STATUSCODE
-      rm ./apidefinitions/${service}-swagger-${PREVIOUS_MINOR_BUILD}.json
+    echo $PREVIOUS_MINOR_BUILD
+    if ! compatible "${service}" "${PREVIOUS_MINOR_BUILD}"; then
+      INCOMPATIBLE_CHANGES+=("$service:$PREVIOUS_MINOR_BUILD")
     fi
   fi
 done
-set -e
-IFS=$Field_Separator
-$INTEGCB_LOCATION/.deps/bin/docker-compose --compatibility up --remove-orphans swagger-diff | tee swagger-diff.out
-grep "swagger diff finished succesfully" swagger-diff.out
-swaggerdiffresult=$?
-
-echo -e "\n\033[1;96m--- Kill running Swagger Diff containers\033[0m\n"
-$INTEGCB_LOCATION/.deps/bin/docker-compose --compatibility down --remove-orphans
-
-$INTEGCB_LOCATION/.deps/bin/docker-compose --compatibility up --remove-orphans swagger-validation | tee swagger-validation-result.out
-grep "swagger validation finished succesfully" swagger-validation-result.out
-swaggervalidationresult=$?
 
-echo -e "\n\033[1;96m--- Kill running Swagger Validation containers\033[0m\n"
-$INTEGCB_LOCATION/.deps/bin/docker-compose --compatibility down --remove-orphans
+NUM_INCOMPATIBLE_CHANGES=${#INCOMPATIBLE_CHANGES[@]}
+if (( NUM_INCOMPATIBLE_CHANGES > 0 )); then
+  # Report any incompatible changes detected.
+  echo
+  echo "Incompatible changes:"
+  for incompat in "${INCOMPATIBLE_CHANGES[@]}"; do
+    echo "- $incompat"
+  done
+  if (( NUM_INCOMPATIBLE_CHANGES > 255 )); then
+    NUM_INCOMPATIBLE_CHANGES=255
+  fi
+fi
 
-exit $swaggerdiffresult && $swaggervalidationresult
+exit $NUM_INCOMPATIBLE_CHANGES
```

Comment:
Out of curiosity, why the need for the funny exit code? Code 0 means success, codes 1...127 are regular errors, while codes C 128...255 typically denote that the process exited due to catching / receiving signal number (C-128). (Only a subset of the latter range is valid as per the permitted signal codes; see `man 7 signal`.) Quoting the `bash` manual:
```
       For the shell's purposes, a command which exits with a zero exit status has succeeded.  An
       exit  status of zero indicates success.  A non-zero exit status indicates failure.  When a
       command terminates on a fatal signal N, bash uses the value of 128+N as the exit status.

       If a command is not found, the child process created to execute it  returns  a  status  of
       127.  If a command is found but is not executable, the return status is 126.
```

Reply:
I think the important thing is that we should not return 0. This is coming from the thunderhead codebase
https://github.com/hortonworks/cloudbreak/pull/11329/files#diff-ffa4a6b378a979862d274632d729be771ea467e80c00fe5f19fdb4048f6c567fR111

Reply:
Okay, but sticking to just two values (0 and 1, say) would suffice. I suggest replacing lines 111..113
```
  if (( NUM_INCOMPATIBLE_CHANGES > 255 )); then
    NUM_INCOMPATIBLE_CHANGES=255
  fi
```
with
```
  NUM_INCOMPATIBLE_CHANGES=1
```

Reply:
done